Sources/MapboxMaps/Camera/BasicCameraAnimator2.swift

```diff
@@ -0,0 +1,105 @@
+import UIKit
+
+public final class BasicCameraAnimator: NSObject, CameraAnimator, CameraAnimatorProtocol {
```

Comment:
What is the reasoning to create this wrapper? 

Is it to preserve backwards compatibility? I wonder if converting `BasicCameraAnimator` to a protocol could work in this case 🤔.

Reply:
This change will allow isolating `CameraAnimationsManager` and `BasicCameraAnimatorImpl` in unit tests (will come in the next PR). Without this change, testing `CameraAnimationsManager` will be difficult since we wouldn't be able to use mocks for `BasicCameraAnimator`.

For internal code, yes, we'd probably just write `CameraAnimationsManager` in terms of `BasicCameraAnimatorProtocol`, but as we push toward v11, I'm recommending that we align on only making protocols public in situations where we would expect end-developers to create their own implementations. Sticking to concrete types in other cases gives us more flexibility to evolve the API without major version bumps.

Reply:
Thanks for the great explanation!